server/routes/api.js

```diff
@@ -0,0 +1,77 @@
+const express = require('express')
+const router = express.Router()
+const data = require('../../expenses')
+const moment = require('moment')
+const Expense = require('../models/Expense')
+
+// data.forEach(d => {
+//     // let date = moment(d.date).format("dddd, MMMM Do YYYY, h:mm:ss a")
+//     let expense = new Expense({
+//         name: d.item,
+//         amount: d.amount,
+//         date: d.date,
+//         group: d.group
+//     })
+//     expense.save()
+// })
+
+router.get('/expenses', function (req, res) {
+    const d1 = req.query.d1 || null
+    const d2 = req.query.d2 || null
+    if(d1 && d2){
+        Expense.find({
+            date: {$gte: moment(d1).format("LLLL"),
+            $lte: moment(d2).format("LLLL")}
+        }).sort({date: -1}).exec(function(err, expenses){
+            res.send(expenses)
+        })
+    }else if(d1){
+        Expense.find({ date: {$gte: moment(d1).format("LLLL")}}).sort({date: -1}).exec(function(err, expenses){
+            res.send(expenses)
+        })
+    }else{
+        Expense.find({}).sort({date: -1}).exec(function(err, expenses){
+            res.send(expenses)
+        })
+    }
+})
+
+router.post('/expense', function(req, res){
+    const data = req.body
+    const date = data.date ? moment(data.date).format("LLLL") : moment().format("LLLL")
```

Comment:
better do something like this -
`let { date } = req.body`
`date = moment(date ? date : undefined).format("LLLL")`